ewatercycle/models/pcrglobwb.py

```diff
@@ -0,0 +1,152 @@
+import shutil
+import time
+from os import PathLike
+from pathlib import Path
+from typing import Any, Iterable, Optional, Tuple
+
+import numpy as np
+import xarray as xr
+from cftime import num2date
+from grpc4bmi.bmi_client_docker import BmiClientDocker
+from grpc4bmi.bmi_client_singularity import BmiClientSingularity
+
+from ewatercycle import CFG
+from ewatercycle.models.abstract import AbstractModel
+from ewatercycle.parametersetdb.config import CaseConfigParser
+
+
+class PCRGlobWB(AbstractModel):
+    """eWaterCycle implementation of PCRGlobWB hydrological model.
+
+    Attributes
+        bmi (Bmi): GRPC4BMI Basic Modeling Interface object
+    """
+    def setup(  # type: ignore
+            self,
+            input_dir: PathLike,
+            cfg_file: PathLike,
```

Comment:
should this be cfg_file_template?

Reply:
I think in most cases users will just pass their own cfg_file, which they don't consider as template, but rather as the *actual* cfg they want to use. Also this is more in line with PyMT which allows you to pass a cfg_file as well. Once we have a nice and complete template, we should set this as the default (so no need to pass a template via setup), and whatever the user passes as cfg_file just overrides the defaults.

So I think cfg_file is fine, but if you insist I will change it.

Reply:
Where did you see in pymt that cfg_file can be passed to setup? I only see path which is same as our work_dir arg.

```python
from pymt.models import Hydrotrend

??Hydrotrend.setup
Signature: Hydrotrend.setup(self, *args, **kwds)
Source:   
    def setup(self, *args, **kwds):
        """Set up a simulation.

        Parameters
        ----------
        path : str, optional
            Path to a folder to set up the simulation. If not given,
            use a temporary folder.

        Returns
        -------
        str
            Path to the folder that contains the set up simulation.
        """
        if len(args) == 0:
            dir_ = tempfile.mkdtemp()
        else:
            dir_ = args[0]

        self._parameters.update(kwds)

        FileSystemLoader(self.datadir).stage_all(dir_, **self._parameters)

        config = self._meta.run["config_file"]
        if config["contents"] and not config["path"]:
            config["path"] = dir_

        if config["path"]:
            with as_cwd(dir_):
                config_file = FileTemplate.write(
                    config["contents"], config["path"], **self._parameters
                )
        else:
            config_file = None

        return config_file, os.path.abspath(dir_)
File:      ~/miniconda38/envs/pymt/lib/python3.9/site-packages/pymt/framework/bmi_setup.py
Type:      function
```

Reply:
@sverhoeven Check, you're right. It says "Here, we didn’t use the outputs from setup because the run directory has been specified, and the configuration file is created within it."

Reply:
Thanks @sverhoeven! In any case I propose we let this be for now. Let's plan a design session to come up with a good way to do this in general.